s3path.py

```diff
@@ -576,6 +574,17 @@ def replace(self, target):
         """
         return self.rename(target)
 
+    def unlink(self, missing_ok=False):
+        """
+        Remove this key from its bucket.
+        If the path is a bucket, use rmdir() instead.
+        """
+        try:
+            self._accessor.unlink(self)
+        except FileNotFoundError:
+            if not missing_ok:
+                raise
```

Comment:
Like we commented in the accessor method this method won't catch the exception that will be raised from boto3...

Also you are missing here some validations you can see example from [rmdir method](https://github.com/liormizr/s3path/blob/master/s3path.py#L210)

Another comment is that you can use more the base Path class to call the _accessor code

So in conclusion personally I would have wrote this method like this:
```
    def unlink(self, missing_ok=False):
        """
        Remove this key from its bucket.
        If the path is a bucket, use rmdir() instead.
        """
        self._absolute_path_validation()
        if not self.is_dir() or self.key is None:
            if missing_ok:
                return
            raise FileNotFoundError()
        return super(). unlink()
```

In this case (if we are not touching the accessor unlink method) the user will get the correct pathlib exception
And also the user will get the boto3 exception if raises
for example:
No key will raise FileNotFoundError.
And boto3 permissions error will be raise as well if the user don't have permissions.

What do you think?

Reply:
Thanks for reviewing this and for making these suggestions!

The reason I opted for this approach is threefold:

1. It mirrors the implementation in pathlib:

```python
class _NormalAccessor(_Accessor):
    ...
    unlink = os.unlink

class Path(PurePath):
    ...
    def unlink(self, missing_ok=False):
        """
        Remove this file or link.
        If the path is a directory, use rmdir() instead.
        """
        if self._closed:
            self._raise_closed()
        try:
            self._accessor.unlink(self)
        except FileNotFoundError:
            if not missing_ok:
                raise
```

2. It avoids excessive API calls. Following the example provided, we would make an api call first to check if the path is a directory, then offload to `pathlib.Path` which doesn't bother with the directory check (because you can't use `unlink` on a directory). Calling the API is *really* expensive (I just did a similar implementation for a different object store and excessive API calls were the difference between operations taking _minutes_ or _seconds_). It probably won't make sense to allow the deletion of folders this way so we can maintain parity with the `pathlib` api.

and finally,

3. It allows us to be more pythonic, handling an exception instead of performing many explicit checks.

Good point about the superclass call, we can definitely rely on it after doing the absolute path validation. I'll make some adjustments and see if you are ok with the new changes (plus documentation updates)